routes/land-preview.js

```diff
@@ -0,0 +1,210 @@
+  res.setHeader('Access-Control-Allow-Methods', '*');
+  if (req.method === 'OPTIONS') {
+    res.end();
+  } else if (req.method === 'POST') {
+    const match = req.url.match(/^\/([0-9]+)/);
+    // console.log('callback server 1', req.url, !!match);
+    if (match) {
+      const index = parseInt(match[1], 10);
+      const cb = cbs[index];
+      // console.log('callback server 2', req.url, index, !!cb);
+      if (cb) {
+        delete cbs[index];
+        cb({req, res});
+      } else {
+        res.statusCode = 404;
+        res.end();
+      }
+    } else {
+      res.statusCode = 404;
+      res.end();
+    }
+  } else {
+    res.statusCode = 404;
+    res.end();
+  }
+});
+server.on('error', serverPromise.reject.bind(serverPromise));
+server.listen(PREVIEW_PORT, PREVIEW_HOST, serverPromise.accept.bind(serverPromise));
+})();
+
+const _handleLandPreviewRequest = async (req, res) => {
+  await serverPromise;
+
+  res.setHeader('Access-Control-Allow-Origin', '*');
+  res.setHeader('Access-Control-Allow-Headers', '*');
+  res.setHeader('Access-Control-Allow-Methods', '*');
+
+  const u = new URL(req.url);
+  const spec = (() => {
+    const path = u.pathname.split('/');
+    if (path.length > 3) {
+      const z = match[1];
+      const x = match[2];
+      const y = match[3];
+      return {
+        z,
+        x,
+        y,
+      };
+    }
+  })();
+  const {query = {}} = u;
+  const cache = !query['nocache'];
+  if (spec) {
+    const {z, x, y} = spec;
+    console.log('preview request', {z, x, y});
+    const key = `${z}/${x}/${y}`;
+    const o = cache ? await (async () => {
+      try {
+        return await getObject(
+          bucketNames.preview,
+          key,
+        );
+      } catch(err) {
+        // console.warn(err);
+        return null;
+      }
+    })() : null;
+    const contentType = 'image/png';
+    if (o) {
+      // res.setHeader('Content-Type', o.ContentType || 'application/octet-stream');
+      res.setHeader('Content-Type', contentType);
+      res.end(o.Body);
+    } else {
+      await ticketManager.lock();
+
+      const p = _makePromise()
+      const index = ++cbIndex;
+      cbs[index] = p.accept.bind(p);
+
+      let page;
+      try {
+        // console.log('preview 3');
+        page = await browser.newPage();
+        // console.log('preview 4');
+        page.on('console', e => {
+          console.log(e);
+        });
+        page.on('error', err => {
+          console.log(err);
+        });
+        page.on('pageerror', err => {
+          console.log(err);
+        });
+
+        let timeout;
+        const t = new Promise((accept, reject) => {
+          timeout = setTimeout(() => {
+            reject(new Error('timed out'));
+          }, 10 * 1000);
+        });
+
+        await Promise.race([
+          (async () => {
+            await page.goto(`https://app.webaverse.com/map.html?x=${x}&y=${y}&sw=16&sh=${z}&dw=128&dh=128&dst=http://${PREVIEW_HOST}:${PREVIEW_PORT}/` + index);

```

Comment:
If we are making `z` the zoom level, then it should probably be `sw=${z}` as well.

These arguments control how many meters a tile is, and you probably want it to be a square tile, meaning `sw` and `sh` are equal.